components/Filter/index.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+
+const Filter = ({ results, setFilteredResults }) => {
+  const [filterText, setFilterText] = useState("");
+
+  const handleFilterChange = (e) => {
+    const newFilterText = e.target.value;
+    setFilterText(newFilterText);
+
+    const filteredResults = results.filter((result) =>
+      result.name.toLowerCase().includes(newFilterText.toLowerCase())
+    );
+    setFilteredResults(filteredResults);
+  };
+
+  return (
+    <div>
```

Comment:
I am not sure here, but the `input` without a wrapping `form` element feels weird.